src/equiv.js

```diff
@@ -131,33 +131,53 @@ QUnit.equiv = ( function() {
 		},
 
 		"set": function( b, a ) {
-			var aArray, bArray;
+			var innerEq,
+				outerEq = true;
 
-			aArray = [];
-			a.forEach( function( v ) {
-				aArray.push( v );
-			} );
-			bArray = [];
-			b.forEach( function( v ) {
-				bArray.push( v );
+			if ( a.size !== b.size ) {
+				return false;
+			}
+
+			a.forEach( function( aVal ) {
+				innerEq = false;
+
+				b.forEach( function( bVal ) {
+					if ( innerEquiv( bVal, aVal ) ) {
+						innerEq = true;
+					}
+				} );
+
+				if ( !innerEq ) {
+					outerEq = false;
+				}
```

Comment:
So I tried this approach. However, it seems `has` simply checks the reference for objects. That means if one of the entries has identical objects but they're not the same object, the check will fail. That's not the current behavior and I don't believe it is the behavior we want.